cpp/src/rolling/nth_element.cuh

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/aggregation.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/gather.cuh>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/utilities/bit.hpp>
+
+#include <limits>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf::detail::rolling {
+
+/// Functor to construct gather-map indices for NTH_ELEMENT rolling aggregation.
+/// By definition, the `N`th element is deemed null (i.e. the gather index is set to "nullify")
+/// for the following cases:
+///   1. The window has fewer elements than `min_periods`.
+///   2. N falls outside the window, i.e. N ∉ [-window_size, window_size).
+///   3. `null_handling == EXCLUDE`, and the window has fewer than `N` non-null elements.
+///
+/// If none of the above holds true, the result is non-null. How the value is determined
+/// depends on `null_handling`:
+///   1. `null_handling == INCLUDE`: The required value is the `N`th value from the window's start.
+///       i.e. the gather index is window_start + N (adjusted for negative N).
+///   2. `null_handling == EXCLUDE`: The required value is the `N`th non-null value from the
+//        window's start. i.e. Return index of the `N`th non-null value.
```

Comment:
Can you change this to the standard doxygen style comments with doxygen tags, etc?